### PR TITLE
Chore/test audio bars

### PR DESCRIPTION
**Pull request by majames**
Test (allthethings)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-[![Build Status](https://travis-ci.org/majames/react-audio-vis.svg?branch=master)](https://travis-ci.org/majames/react-audio-vis)
-[![Coverage Status](https://coveralls.io/repos/github/majames/react-audio-vis/badge.svg?branch=chore%2Fcoverage-stats)](https://coveralls.io/github/majames/react-audio-vis?branch=chore%2Fcoverage-stats)
+[![Build Status](https://travis-ci.org/devlucky/react-audio-vis.svg?branch=master)](https://travis-ci.org/devlucky/react-audio-vis)
```
Comment (majames) [inline README.md:1]: Update badges to point at devlucky

File: src/analyser/index.tsx [hunk 2]
```diff
@@ -25,6 +29,26 @@ export class Analyser {
     if (source) { source.disconnect(); }
   }
 
+  getBucketedByteFrequencyData = (maxNumBuckets: number): Array<number> => {
```
Comment (majames) [inline src/analyser/index.tsx:32]: Move logic for bucketing frequency data into `Analyser` (other visualisations can use this)

File: src/bars/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      const analyser = {audioEl} as any;
+
+      const wrapper = shallow(<AudioBars analyser={analyser} />);
+      wrapper.instance().componentDidMount();
+
+      const {addEventListener} = audioEl;
+      expect(addEventListener).toHaveBeenCalledTimes(3);
+      expect(addEventListener.mock.calls[0][0]).toEqual('playing');
+      expect(addEventListener.mock.calls[1][0]).toEqual('pause');
+      expect(addEventListener.mock.calls[2][0]).toEqual('ended');
```
Comment (zzarcon) [inline src/bars/index.test.tsx:19]: Those cases are gold!

File: src/bars/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    });
+
+    it('clears the canvas', () => {
+      const wrapper = shallow(<AudioBars analyser={analyser} />);
+
+      wrapper.instance().canvasContext = canvasContext;
+      wrapper.instance().drawBars();
+
+      expect(canvasContext.clearRect).toHaveBeenCalledTimes(1);
+      expect(canvasContext.clearRect).toHaveBeenLastCalledWith(0 , 0, 400, 400);
```
Comment (zzarcon) [inline src/bars/index.test.tsx]: Not sure if this ```it``` is giving us some value or we are getting to deep here. Like from the outside, seeing ```0 , 0, 400, 400``` looks really weird

File: src/bars/index.test.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    it('draws the correct number of bars to the canvas', () => {
+      const wrapper = shallow(<AudioBars analyser={analyser} />);
+
+      wrapper.instance().canvasContext = canvasContext;
+      wrapper.instance().drawBars();
+
+      const firstBarHeight = 1 / 256 * 400;
+      const firstBarWidth = 400 / 4;
+      expect(canvasContext.fillRect).toHaveBeenCalledTimes(freqData.length);
+      expect(canvasContext.fillRect).toHaveBeenCalledWith(0, 400 - firstBarHeight, firstBarWidth, firstBarHeight);
```
Comment (zzarcon) [inline src/bars/index.test.tsx]: Same thing here as before, I do like the ```draws the correct number of bars to the canvas``` test and I think we should keep it, but I will prefer to see something like:

```javascript

expect(wrapper.instance().drawSingleBar).toHaveBeenCalledTimes(expectedNumberOfDrawedBars);
```

What do you think? I don't think doing such calculations in the testsuite is the way to go, I think we should try to abstract that more

File: src/bars/index.tsx [hunk 6]
```diff
@@ -60,35 +57,22 @@ export class AudioBars extends Component<AudioBarsProps, {}> {
...
-    const bufferLength = dataArray.length;
-
-    // chunk values if too many to display
-    const numValuesPerChunk = Math.ceil(bufferLength / numBarsToDraw);
-    const chunkedData = chunk(dataArray, numValuesPerChunk);
-    const barValues = chunkedData.map((arr: Array<number>) => sum(arr) / arr.length);
-
-    const barWidth = canvasWidth / numBarsToDraw;
+    const maxNumBarsToDraw = 64;
+    const barValues = analyser.getBucketedByteFrequencyData(maxNumBarsToDraw);
```
Comment (zzarcon) [inline src/bars/index.tsx:71]: Nice one

**Review by zzarcon — CHANGES_REQUESTED** [review]

File: src/bars/index.test.tsx (hunk 4 above)
Comment (majames) [reply src/bars/index.test.tsx]: Yeah. I defs think it looks a little weird... but I do like testing the logic for clearing the canvas.

How about I use *explaining variables*? Like:

`const canvasDefaultWidth = 400;`
`const canvasDefaultHeight = 400;`

**Review by zzarcon — APPROVED** [review]

**Pull request merged**